pkg/channel/channel.go

```diff
@@ -21,3 +21,8 @@ type AppChannel interface {
 	GetBaseAddress() string
 	InvokeMethod(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error)
 }
+
```

Comment:
I do not think we need to introduce SetAppChannel interface. Please add `SetAppChannel()` function to each interface.